beacon-chain/rpc/validator/attester.go

```diff
@@ -1,94 +1,44 @@
-package attester
+package validator
 
 import (
 	"context"
 
 	"github.com/gogo/protobuf/proto"
 	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
 	"github.com/prysmaticlabs/go-ssz"
-	"github.com/prysmaticlabs/prysm/beacon-chain/blockchain"
 	"github.com/prysmaticlabs/prysm/beacon-chain/cache"
 	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
 	"github.com/prysmaticlabs/prysm/beacon-chain/core/state"
-	"github.com/prysmaticlabs/prysm/beacon-chain/db"
-	"github.com/prysmaticlabs/prysm/beacon-chain/operations/attestations"
-	"github.com/prysmaticlabs/prysm/beacon-chain/p2p"
-	"github.com/prysmaticlabs/prysm/beacon-chain/sync"
-	pb "github.com/prysmaticlabs/prysm/proto/beacon/rpc/v1"
-	"github.com/sirupsen/logrus"
 	"go.opencensus.io/trace"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
-var log logrus.FieldLogger
-
-func init() {
-	log = logrus.WithField("prefix", "rpc/attester")
-}
-
-// Server defines a server implementation of the gRPC Attester service,
-// providing RPC methods for validators acting as attesters to broadcast votes on beacon blocks.
-type Server struct {
-	P2p              p2p.Broadcaster
-	BeaconDB         db.Database
-	AttPool          attestations.Pool
-	AttReceiver      blockchain.AttestationReceiver
-	HeadFetcher      blockchain.HeadFetcher
-	AttestationCache *cache.AttestationCache
-	SyncChecker      sync.Checker
-}
-
-// SubmitAttestation is a function called by an attester in a sharding validator to vote
-// on a block via an attestation object as defined in the Ethereum Serenity specification.
-func (as *Server) SubmitAttestation(ctx context.Context, att *ethpb.Attestation) (*pb.AttestResponse, error) {
-	root, err := ssz.HashTreeRoot(att.Data)
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "Could not tree hash attestation: %v", err)
-	}
-
-	// Broadcast the new attestation to the network.
-	if err := as.P2p.Broadcast(ctx, att); err != nil {
-		return nil, status.Errorf(codes.Internal, "Could not broadcast attestation: %v", err)
-	}
-
-	go func() {
-		ctx = trace.NewContext(context.Background(), trace.FromContext(ctx))
-		attCopy := proto.Clone(att).(*ethpb.Attestation)
-		if err := as.AttPool.SaveUnaggregatedAttestation(attCopy); err != nil {
-			log.WithError(err).Error("Could not handle attestation in operations service")
-			return
-		}
-	}()
-
-	return &pb.AttestResponse{Root: root[:]}, nil
-}
-
-// RequestAttestation requests that the beacon node produce an IndexedAttestation,
-// with a blank signature field, which the validator will then sign.
-func (as *Server) RequestAttestation(ctx context.Context, req *pb.AttestationRequest) (*ethpb.AttestationData, error) {
+// GetAttestationData requests that the beacon node produce an attestation data object,
```

Comment:
Review this file - here we use the named methods from Ethereum APIs pretty much copying over logic we had in our attester RPC server into two functions: `GetAttestationData` and `ProposeAttestation`. No real logic changes.